pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -356,9 +403,40 @@ func NewNetworkPolicyController(kubeClient clientset.Interface,
 					}
 					return groupNames.List(), nil
 				},
+				ServiceIndex: func(obj interface{}) ([]string, error) {
+					acnp, ok := obj.(*secv1alpha1.ClusterNetworkPolicy)
+					if !ok {
+						return []string{}, nil
+					}
+					serviceNames := sets.String{}
+					for _, egressRule := range acnp.Spec.Egress {
+						for _, service := range egressRule.ToService {
+							serviceNames.Insert(fmt.Sprintf("%s/%s", service.Namespace, service.Name))
+						}
+					}
+					return serviceNames.List(), nil
+				},
+				ServiceTypeIndex: func(obj interface{}) ([]string, error) {
+					acnp, ok := obj.(*secv1alpha1.ClusterNetworkPolicy)
+					if !ok {
+						return []string{}, nil
+					}
+					serviceTypes := sets.String{}
+					for _, egressRule := range acnp.Spec.Egress {
+						for _, serviceReference := range egressRule.ToService {
+							service, err := n.serviceLister.Services(serviceReference.Namespace).Get(serviceReference.Name)
```

Comment:
This is expense for an index function. At least mention that it will only be invoked on Node ADD/UPDATE/DELETE which is quite unoften.

Reply:
Added comments.